### DISCUSSION TITLE
choose from a long list

### DISCUSSION THREAD

**Discussion by electriquo** [charmbracelet/gum#378] [Q&A]
using `choose`, gum displays each item in a new line, leaving most of the line empty. when feeding choose with a long list of items, (depending on how many columns your terminal can display) the header and some items are not shown.

is there a way to make gum utilize the terminal real-estate and display the items in rows and columns?

**Comment by beltagyy** [comment]
gum choose currently renders items in a single column. There is no built-in flag to switch to a multi-column layout.\n\nFor long lists, the best alternatives are:\n1. Use gum filter instead. Users can type to narrow down a long list, which avoids scrolling entirely.\n2. If you need multiple visible choices, pipe your list through column or fzf before passing to gum choose for final selection.\n\nIf you specifically need a grid selector inside gum, that would require a new feature request in the Gum repository since it is not supported by the underlying choose model today.
